src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,95 @@
+package duke.storage;
+
+import duke.task.*;
+import duke.DukeException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.Scanner;
+
+public class Storage {
+    private File file;
+
+    private static final String ERROR_FAILED_TO_READ     = "Failed to read save data. Creating new task list.";
+    private static final String ERROR_FAILED_SAVE        = "Failed to save file.";
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("EEE, d MMM uuuu, hh.mma");
+
+    public Storage(String filePath) {
+        file = new File(filePath);
+    }
+
+    public String getFilePath() {
+        return file.getAbsolutePath();
+    }
+
+    public TaskList load() throws DukeException {
+        TaskList tasks = new TaskList();
+        try {
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNext()) {
+                String[] input = sc.nextLine().split(" [|] ");
+                Task task;
+                switch (input[0]) {
+                case "T":
+                    task = new Todo(input[2], input[1].equals("1"));
+                    break;
+                case "D":
+                    task = new Deadline(input[2], LocalDateTime.parse(input[3], DATE_TIME_FORMATTER),
+                            input[1].equals("1"));
+                    break;
+                case "E":
+                    task = new Event(input[2], LocalDateTime.parse(input[3], DATE_TIME_FORMATTER),
+                            input[1].equals("1"));
+                    break;
+                default:
+                    throw new DukeException(ERROR_FAILED_TO_READ);
+                }
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException | DukeException | ArrayIndexOutOfBoundsException
```

Comment:
Perhaps the exceptions could be split up so that they throw DukeExceptions with different error messages to make it clearer to the user what and where the error occured!